### PR TITLE
Add params to prometheus_simple

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add support for passing params to the prometheus scrape config.

**Testing:**
Unit tests.

**Documentation:** 
Added to README and testdata.

### CODE REVIEW

**Comment by atoulme** [comment]
I tested this independently in a pet project, and it works well there.

**Review by rmfitzpatrick — APPROVED** [review]

File: receiver/simpleprometheusreceiver/README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ The following settings are optional:
 - `collection_interval` (default = `10s`): The internal at which metrics should
 be emitted by this receiver.
 - `metrics_path` (default = `/metrics`): The path to the metrics endpoint.
+- `params` (default = `{}`): the parameters to pass to the metrics endpoint.
```
Comment (tigrannajaryan) [inline receiver/simpleprometheusreceiver/README.md:22]:
```suggestion
- `params` (default = `{}`): the query parameters to pass to the metrics endpoint. If specified, params are appended to `metrics_path` to form the URL with which the target is scraped.
```

**Comment by jrcamp** [comment]
@atoulme lgtm

File: receiver/simpleprometheusreceiver/README.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ The following settings are optional:
 - `collection_interval` (default = `10s`): The internal at which metrics should
 be emitted by this receiver.
 - `metrics_path` (default = `/metrics`): The path to the metrics endpoint.
+- `params` (default = `{}`): the query parameters to pass to the metrics endpoint. If specified, params are appended to `metrics_path` to form the URL with which the target is scraped.
```
Comment (tanner-bruce) [inline receiver/simpleprometheusreceiver/README.md:22]:
```suggestion
- `params` (default = `{}`): The query parameters to pass to the metrics endpoint. If specified, params are appended to `metrics_path` to form the URL with which the target is scraped.
```

File: receiver/simpleprometheusreceiver/config.go [hunk 3]
```diff
@@ -30,6 +31,8 @@ type Config struct {
 	CollectionInterval time.Duration `mapstructure:"collection_interval"`
 	// MetricsPath the path to the metrics endpoint.
 	MetricsPath string `mapstructure:"metrics_path"`
+	// Params the parameters to the metrics endpoint.
```
Comment (tanner-bruce) [inline receiver/simpleprometheusreceiver/config.go:34]:
```suggestion
	// Params are the parameters to pass to the metrics endpoint.
```

File: receiver/simpleprometheusreceiver/testdata/config.yaml [hunk 4]
```diff
@@ -4,6 +4,8 @@ receivers:
     endpoint: "localhost:1234"
     collection_interval: 30s
     metrics_path: /v2/metrics
+    params:
+      columns: "name,messages"
```
Comment (tanner-bruce) [inline receiver/simpleprometheusreceiver/testdata/config.yaml:8]:
```suggestion
      columns: [name, messages]
```

**Review by tanner-bruce — COMMENTED** [review]
Nice stuff! A couple minor nits.

File: receiver/simpleprometheusreceiver/config.go (hunk 3 above)
Comment (atoulme) [reply receiver/simpleprometheusreceiver/config.go:34]: No, this is not how go renders docs. See the rest of the doc - or ScrapeConfig where I lifted this.

File: receiver/simpleprometheusreceiver/testdata/config.yaml (hunk 4 above)
Comment (atoulme) [reply receiver/simpleprometheusreceiver/testdata/config.yaml:8]: OK, that's not a good example config - I'll change it, but your suggestion breaks tests. I'll have to rework this a bit.

Comment (atoulme) [reply receiver/simpleprometheusreceiver/testdata/config.yaml:8]: Added one more parameter - both comma-delimited and arrays are supported.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**